src/aead/aead.rs

```diff
@@ -25,6 +25,82 @@
 //! [AEAD]: http://www-cse.ucsd.edu/~mihir/papers/oem.html
 //! [`crypto.cipher.AEAD`]: https://golang.org/pkg/crypto/cipher/#AEAD
 
+//!
+//! **Example (if you don't know what you are doing, read this first)**
+//!
+//!```rust
+//!
+//!extern crate ring;
+//!
+//!use ring::aead::*;
+//!use ring::pbkdf2::*;
+//!use ring::rand::SystemRandom;
+//!
+//!fn main() {
+//!    // The password will be used to generate a key and it is normally used when
+//!    // it is generated by the user. If you need a key which will be stored in a file you
+//!    // generate it directly.
+//!    let password = b"nice password";
+//!
+//!    // Usually the salt has some random data and something that relates to the user
+//!    // like an username and it must be 8 bytes (64 bits)
+//!    let salt = [0, 1, 2, 3, 4, 5, 6, 7];
+//!
+//!    // Keys must have 32 bytes and if you want to use an user password for it you have
+//!    // to use an algorithm to derive it (in this case pbkdf2).
+//!    let mut key = [0; 32];
+//!    derive(&HMAC_SHA256, 100, &salt, &password[..], &mut key);
+//!
```

Comment:
Brian wrote:

> I don't think we should teach people how to derive a key from a password in this example. This example scheme is actually not as secure as we'd like it to be w.r.t. the nonce handling.

I can't find that comment anymore here. Without understaind his exact reasons (again, I'm not a crypto expert, so I can't judge), I'd say the example code shouldn't demonstrate the derivation if it's not safe.

OTOH: Yes, it is a common problem, and it's common to derive a secret from a password and then encrypt something with it. So eventually, this (upper) part of the sample code will be relevant -- don't throw it away. I hope that *ring* will soon offer a safe solution to do that? Maybe it has to do with nonce-misuse-resistant ciphers? https://github.com/briansmith/ring/issues/413

Once the key derivation sample becomes relevant, it can be debated whether it should be placed here or elsewhere. Either way, I think it should not be in this same *block*  of code.